source/core/constants.ts

```diff
@@ -58,3 +58,20 @@ export const kyOptionKeys: KyOptionsRegistry = {
 	onDownloadProgress: true,
 	fetch: true,
 };
+
+export const requestOptions = new Set([
```

Comment:
Is there any other way than having a hard-coded list here? We'll have to keep this up to date and if we miss something, it could cause problems.

Reply:
I really want to avoid this too. Let me give it some thought.

Reply:
Maybe a similar solution to the `KyOptionsRegistry` can be employed here to defer the list keeping to Typescript and undici. I will try this.

Reply:
So there are two RequestInit interfaces that are relevant here: the one exported from node's `undici-types` and the one exported from Typescript's `lib.dom.d.ts`.


### Common fields present in both interfaces:
- `method`
- `keepalive`
- `headers`
- `body`
- `redirect`
- `integrity`
- `signal`
- `credentials`
- `mode`
- `referrer`
- `referrerPolicy`
- `window`

### Unique to `undici-types`:
- `dispatcher`
- `duplex`

### Unique to Typescript's `lib.dom.d.ts`:
- `cache`


### Standard Request Options [as specified in MDN](https://developer.mozilla.org/en-US/docs/Web/API/Request/Request#options)

#### Present in both `undici-types` and `lib.dom.d.ts`:
- `method`
- `headers`
- `body`
- `mode`
- `credentials`
- `redirect`
- `referrer`
- `referrerPolicy`
- `integrity`
- `keepalive`
- `signal`

#### Present only in `lib.dom.d.ts`:
- `cache`

#### Missing in both `undici-types` and `lib.dom.d.ts`:
- `priority` (⚠️ this field is still considered experimental but it is already supported in Chromium based browsers)

With all of this being said, I think we can leave `priority` out and let it be detected as an unknown property since it will still be passed down to fetch.

Please let me know what you think of this approach @sindresorhus and @sholladay.

Reply:
Yeah, that sounds like a good approach.